avocado_vt/plugins/vt_joblock.py

```diff
@@ -0,0 +1,119 @@
+import errno
+import logging
+import os
+import re
+import random
+import string
+import sys
+
+from avocado.core import exit_codes
+from avocado.core.settings import settings
+from avocado.plugins.base import JobPre, JobPost
+
+from ..test import VirtTest
+
+
+class VTJobLock(JobPre, JobPost):
+
+    name = 'vt-joblock'
+    description = 'Avocado-VT Job Lock/Unlock'
+
+    def __init__(self):
+        self.log = logging.getLogger("avocado.app")
+        self.has_vt_test = False
+        self.lock_dir = settings.get_value(section="plugins.vtjoblock",
+                                           key="dir",
+                                           key_type=str,
+                                           default='/tmp')
```

Comment:
I like the idea of using the `data_dir`. The reason is that, **implicitly**, this is what we're trying to lock.

But the discussed goal, was to prevent parallel executions (period).  From the card:

"Avocado should implement some locking around virt-tests. For now, maybe a big VTLOCK that prevents parallel vt jobs would be enough, no need to be fine grained."

And then, if the user decides to tweak the environment to allow parallel runs on the same system (same data dir or not) is is own business. That's why this code is modeled like that, and even the documentation gives it away that admins/users can set this to their data_dirs. This lock in the `data_dir` would effectively mean a slightly finer grain lock, at the data_dir level, and not system-wide level. It's nice, but, IMHO out of the scope and slightly out of the described expected outcome for this task.

Then, what I dislike: trying to be too smart.  If a stale lock is left on the system, say, because the system shut down or crashed, then the user will get the message about the job that seems to be running with a given PID.  It seems enough at this point.  The user can check if that job is still running or not, and clean up the lock file if they're sure it's a stale one.  If we decide to check if the process is still running ourselves during job runtime, and then cleaning up the PID file automatically, we incur into so many possible error/race conditions.

Again, this is supposed to be the simplest possible, but effective,  solution.  What you're describing is much more elaborated.  Not that I don't like more complete solutions, it just looks like it's out of the task's scope.

If you think any of your suggestions is necessary to deliver an *effective* solution, then let's deal with them.  But if you're proposing better, as in more elaborate, feature rich, solutions, then I think it's out of the scope of this task.